hedera-node/src/main/java/com/hedera/services/ledger/interceptors/StakeAwareAccountsCommitsInterceptor.java

```diff
@@ -104,19 +112,21 @@ public void preview(final EntityChangeSet<AccountID, MerkleAccount, AccountPrope
 
 		// Once rewards are activated, they remain activated
 		rewardsActivated = rewardsActivated || networkCtx.get().areRewardsActivated();
-		// Initialize funding reward balance (will only be updated and consulted if rewards are not active)
+		// Will only be updated and consulted if rewards are not active
 		newFundingBalance = -1;
 
 		// Iterates through the change set, maintaining two invariants:
-		//   1. At the beginning of iteration i, any account in the [0, i) range that is reward-able due to
-		//      change in balance, stakedAccountId, stakedNodeId, declineRewards fields; _has_ been rewarded.
+		//   1. At the beginning of iteration i, any account in the [0, i) range that was reward-able due to
+		//      a change in balance, stakedAccountId, stakedNodeId, or declineRewards fields has been rewarded.
+		//      (IMPORTANT: this reward could be zero if the effective declineRewards is true.)
 		//   2. Any account whose stakedToMe balance was affected by one or more changes in the [0, i) range
-		//      has been, if not already present, added to the pendingChanges; and its changes reflect all
-		//      these stakedToMe change
-		updateAccountStakes(pendingChanges);
-		// Iterates through the change set to update node stakes; requires a separate loop so all stakedToMe are final
-		updateNodeStakes(pendingChanges);
-		finalizeRewardBalanceChange(pendingChanges);
+		//      has been, if not already present, added to the pendingChanges; and its updated stakedToMe is
+		//      reflected in stakedToMeUpdates.
+		updateRewardsAndElections(pendingChanges);
+		// Updates node stakes and constructs any finishers the ledger will use to set stakedToMe and
+		// stakePeriodStart fields on the mutable account instances
+		finalizeStakeMetadata(pendingChanges);
+		finalizeRewardBalance(pendingChanges);
```

Comment:
Just observed , on line 133 this condition should be having `>=`

```
if (!rewardsActivated && newFundingBalance >= dynamicProperties.getStakingStartThreshold()) {

Reply:
Fixed.